src/orion/core/worker/storage_backup.py

```diff
@@ -0,0 +1,473 @@
+    src_storage,
+    dst_storage,
+    resolve,
+    experiments,
+    import_benchmarks=True,
+    progress_callback=None,
+):
+    """Prepare importation.
+
+    Compute all changes to apply to make import and return changes as dictionaries.
+
+    Parameters
+    ----------
+    src_storage: BaseStorageProtocol
+        storage to import from
+    dst_storage: BaseStorageProtocol
+        storage to import into
+    resolve:
+        resolve policy
+    experiments:
+        experiments to import from src_storage into dst_storage
+    import_benchmarks:
+        if True, benchmarks will be also imported from src_database
+    progress_callback:
+        See :func:`load_database`
+
+    Returns
+    -------
+    A couple (queries to delete, data to add) representing
+    changes to apply to dst_storage to make import
+    """
+    assert resolve in ("ignore", "overwrite", "bump")
+
+    queries_to_delete = {}
+    data_to_add = {}
+
+    if import_benchmarks:
+        src_benchmarks = src_storage.fetch_benchmark({})
+        for i, src_benchmark in enumerate(src_benchmarks):
+            _describe_import_progress(
+                STEP_CHECK_BENCHMARKS, i, len(src_benchmarks), progress_callback
+            )
+            dst_benchmarks = dst_storage.fetch_benchmark(
+                {"name": src_benchmark["name"]}
+            )
+            if dst_benchmarks:
+                (dst_benchmark,) = dst_benchmarks
+                if resolve == "ignore":
+                    logger.info(
+                        f'Ignored benchmark already in dst: {src_benchmark["name"]}'
+                    )
+                    continue
+                if resolve == "overwrite":
+                    logger.info(
+                        f'Overwrite benchmark in dst, name: {src_benchmark["name"]}'
+                    )
+                    queries_to_delete.setdefault(COL_BENCHMARKS, []).append(
+                        {"_id": dst_benchmark["_id"]}
+                    )
+                elif resolve == "bump":
+                    raise DatabaseError(
+                        "Can't bump benchmark version, "
+                        "as benchmarks do not currently support versioning."
+                    )
+            # Delete benchmark database ID so that a new one will be generated on insertion
+            del src_benchmark["_id"]
+            data_to_add.setdefault(COL_BENCHMARKS, []).append(src_benchmark)
+        _describe_import_progress(
+            STEP_CHECK_BENCHMARKS,
+            len(src_benchmarks),
+            len(src_benchmarks),
+            progress_callback,
+        )
+
+    _describe_import_progress(STEP_CHECK_DST_EXPERIMENTS, 0, 1, progress_callback)
+    all_dst_experiments = dst_storage.fetch_experiments({})
+    # Dictionary mapping dst exp name to exp version to list of exps with same name and version
+    dst_exp_map = {}
+    last_experiment_id = max((data["_id"] for data in all_dst_experiments), default=0)
```

Comment:
The `_id`s may not all be integers. It won't be the case if the `dst_storage` is mongodb for example.

Reply:
Ok ! Code does not rely on ID as integer anymore.